src/Symfony/Component/DependencyInjection/Attribute/When.php

```diff
@@ -20,6 +20,9 @@
 class When
 {
     public function __construct(
+        /**
+         * The environment under which the class will be registered as a service (i.e. "dev", "staging", "prod").
```

Comment:
I'm not sure "staging" is frequently used. It's not in the documentation. "test" is documented and have special behavior in DotEnv.

```suggestion
         * The environment under which the class will be registered as a service (i.e. "dev", "test", "prod").
```